managed/src/main/java/com/yugabyte/yw/commissioner/tasks/KubernetesTaskBase.java

```diff
@@ -486,6 +501,10 @@ public void createSingleKubernetesExecutorTaskForServerType(
     }
     if (config != null) {
       params.config = config;
+      // This assumes that the config is az config.
+      // params.namespace remains null if config is not passed.
+      params.namespace = PlacementInfoUtil.getKubernetesNamespace(taskParams().nodePrefix,
```

Comment:
minor: would be good to add a comment at the start of this function that az is null for single zone providers, it is a bit confusing.